core/org.eclipse.birt.core/src/org/eclipse/birt/core/util/IOUtil.java

```diff
@@ -158,10 +158,17 @@ public final static void writeShort(OutputStream outputStream, short value) thro
 	 * @throws IOException
 	 */
 	public final static int readInt(InputStream inputStream) throws IOException {
-		int ch1 = inputStream.read();
-		int ch2 = inputStream.read();
-		int ch3 = inputStream.read();
-		int ch4 = inputStream.read();
+		byte[] bytebuffer = new byte[4];
```

Comment:
I think one should take DataInputStream.readInt() as a template.
Note that DataInputStream uses only one static 8 bytes buffer, while your code creates a new buffer on the heap *for each call* and then calls ByteArray.getInt to convert the buffer to an int.